modules/core/include/opencv2/core/eigen.hpp

```diff
@@ -47,6 +47,10 @@
 
 #include "opencv2/core.hpp"
 
+#if EIGEN_WORLD_VERSION >= 3 && EIGEN_MAJOR_VERSION >= 3
```

Comment:
Condition is wrong - Eigen 4.2 will not pass.
But probably they would move this code from "unsupported" part, so it is better to stick with Eigen 3.x only for now (`EIGEN_WORLD_VERSION == 3`).

---

It is better to avoid condition duplication - define and use separate macro for this feature:
```
#if <condition>
#include <unsupported/Eigen/CXX11/Tensor>
#define OPENCV_EIGEN_TENSOR_SUPPORT
#endif

...

#ifdef OPENCV_EIGEN_TENSOR_SUPPORT
... tensor code or tests ...
#endif
```

Reply:
I fixed the condition duplication as you suggested and restricted to Eigen 3.

I'm confused about why you think the condition is wrong though. Eigen v3.2 (I assume a typo was made when you wrote `4.2`) does not include Eigen Tensor so it should not pass. Eigen Tensor was first introduced and packaged into Eigen v3.3.0 which implies

```
EIGEN_WORLD_VERSION = 3
EIGEN_MAJOR_VERSION = 3
EIGEN_MINOR_VERSION = 0
```
I left out the check for `EIGEN_MINOR_VERSION` because it is unnecessary. Am I missing something?